scylla-cql/src/types/deserialize/value.rs

```diff
@@ -242,6 +292,69 @@ mod tests {
         compat_check::<f64>(&ColumnType::Double, make_bytes(&123f64.to_be_bytes()));
         compat_check::<f64>(&ColumnType::Double, make_bytes(&(-123f64).to_be_bytes()));
+
+        // big integers
+        const PI_STR: &[u8] = b"3.1415926535897932384626433832795028841971693993751058209749445923";
+        let num1 = PI_STR[2..].to_vec();
+        let num2 = vec![b'-']
+            .into_iter()
+            .chain(PI_STR[2..].iter().copied())
+            .collect::<Vec<_>>();
+        let num3 = b"0".to_vec();
+
+        // native - CqlVarint
+        {
+            let num1 = CqlVarint::from_signed_bytes_be(num1.clone());
+            let num2 = CqlVarint::from_signed_bytes_be(num2.clone());
+            let num3 = CqlVarint::from_signed_bytes_be(num3.clone());
+            compat_check_serialized::<CqlVarint>(&ColumnType::Varint, &num1);
+            compat_check_serialized::<CqlVarint>(&ColumnType::Varint, &num2);
+            compat_check_serialized::<CqlVarint>(&ColumnType::Varint, &num3);
+        }
+
+        #[cfg(feature = "num-bigint-03")]
+        {
+            use num_bigint_03::BigInt;
+
+            let num1 = BigInt::parse_bytes(&num1, 10).unwrap();
+            let num2 = BigInt::parse_bytes(&num2, 10).unwrap();
+            let num3 = BigInt::parse_bytes(&num3, 10).unwrap();
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num1);
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num2);
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num3);
+        }
+
+        #[cfg(feature = "num-bigint-04")]
+        {
+            use num_bigint_04::BigInt;
+
+            let num1 = BigInt::parse_bytes(&num1, 10).unwrap();
+            let num2 = BigInt::parse_bytes(&num2, 10).unwrap();
+            let num3 = BigInt::parse_bytes(&num3, 10).unwrap();
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num1);
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num2);
+            compat_check_serialized::<BigInt>(&ColumnType::Varint, &num3);
+        }
+
+        // big decimals
+        #[cfg(feature = "bigdecimal-04")]
+        {
+            use bigdecimal_04::BigDecimal;
+
+            let num1 = PI_STR.to_vec();
+            let num2 = vec![b'-']
+                .into_iter()
+                .chain(PI_STR.iter().copied())
+                .collect::<Vec<_>>();
+            let num3 = b"0.0".to_vec();
+
+            let num1 = BigDecimal::parse_bytes(&num1, 10).unwrap();
+            let num2 = BigDecimal::parse_bytes(&num2, 10).unwrap();
+            let num3 = BigDecimal::parse_bytes(&num3, 10).unwrap();
+            compat_check_serialized::<BigDecimal>(&ColumnType::Decimal, &num1);
+            compat_check_serialized::<BigDecimal>(&ColumnType::Decimal, &num2);
+            compat_check_serialized::<BigDecimal>(&ColumnType::Decimal, &num3);
+        }
```

Comment:
No idea.

Reply:
> Why some types are only tested in this function and some are both tested here and in a separate function?

I'm not entirely sure because it was a long time ago, so what I'll write here might be wrong (with more probability than usual, at least).

There are two kinds of tests in this file:

1. Tests which verify that the `DeserializeCql` works properly (this is new code which needs to be tested)
2. Tests which verify that the `DeserializeCql` implementation agrees with the legacy `FromCqlVal`-based one

They both serve a different purpose.

Reply:
My main question still remains: why are some types only tested in the second way and not the first way?

Reply:
I don't remember, sorry. I might have just forgot to write the first kind of test for some types.

Reply:
Ok, unless there is a reason for some types to be absent from one type of tests please implenent both @wprzytula